### PR TITLE
Changes from trying to generate updated database

### PR DESCRIPTION
**Pull request by mithro**
* `074-dump_all` - Don't fail if output dir doesn't exist.
 * Add Python virtualenv to make it easier to install all the Python requirements.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+env
```
Comment (galv) [inline .gitignore]: Change to `/env/` to avoid capturing any non-directory env files that aren't at the top level.

File: Makefile [hunk 2]
```diff
@@ -2,6 +2,11 @@ CLANG_FORMAT ?= clang-format
 
 .PHONY: database format clean
 
+env:
```
Comment (galv) [inline Makefile]: Make this rule .PHONY? Up to you. Note that it won't work if something breaks before the last line. Alternatively, make a directory like `env-in-progress` and `mv` it to a directory called `env`.

File: Makefile [hunk 3]
```diff
@@ -2,6 +2,11 @@ CLANG_FORMAT ?= clang-format
...
+	virtualenv --python=python3 env
+	. env/bin/activate; pip install -r requirements.txt
+	ln -s $(PWD)/prjxray env/lib/python3.*/site-packages/
```
Comment (galv) [inline Makefile]: Change to `ln -sf` to make this make rule idempotent.

File: .gitignore (hunk 1 above)
Comment (mithro) [reply .gitignore]: Done.

File: Makefile (hunk 2 above)
Comment (mithro) [reply Makefile]: Done.

File: Makefile (hunk 3 above)
Comment (mithro) [reply Makefile]: Done.

**Comment by mithro** [comment]
Hi @galv,

Thanks for helping review this code, but who are you?

Tim @mithro Ansell

**Comment by galv** [comment]
Hi @mithro. I'm a friend of @mcmasterg. I saw that the currently uploaded database was months old recently, but then I saw your PR which sounds like it'll hopefully update, so I figured I'd pitch in a little bit.

**Comment by mithro** [comment]
@galv Ahh cool! Yes, there should be a database update soon.

**Pull request merged**